combo_cases_and_deaths/delphi_combo_cases_and_deaths/run.py

```diff
@@ -86,15 +86,49 @@ def compute_special_geo_dfs(df, signal, geo):
     return df
 
 
-def combine_usafacts_and_jhu(signal, geo, date_range, issue_range=None, fetcher=covidcast.signal):
-    """Add rows for PR from JHU signals to USA-FACTS signals.
+def merge_dfs_by_geos(usafacts_df, jhu_df, geo_to_fetch):
+    """Combine the queried usafacts and jhu dataframes based on the geo type."""
+    # State level
+    if geo_to_fetch == 'state':
+        combined_df = maybe_append(
+            usafacts_df,
+            jhu_df if jhu_df is None else jhu_df[jhu_df["geo_value"] == 'pr']) # add territories
+    # County level
+    elif geo_to_fetch == 'county':
+        combined_df = maybe_append(
+            usafacts_df,
+            jhu_df if jhu_df is None else jhu_df[jhu_df["geo_value"].str.startswith("72")])
+    # For MSA and HRR level, they are the same
+    elif geo_to_fetch == 'msa':
+        # df = GMPR._load_crosswalk("fips", "msa")
+        # puerto_rico_fips = ["72" + str(i).zfill(3) for i in range(999)]
+        # puerto_rico_msas = df[df["fips"].isin(puerto_rico_fips)]["msa"].unique()
+        puerto_rico_msas = [
+            '10380',
+            '11640',
+            '25020',
+            '32420',
+            '38660',
+            '41900',
+            '41980',
+            '49500'
+        ]
```

Comment:
what's going on here? i'm not super hyped about hard-coded MSAs, since if we ever need to fix them in the geo utils then we'll have to remember to fix them here too

Reply:
(would it be easier/faster/clearer to have GMPR just add a fips column to the msa data, then filter by fips?)

Reply:
yea, so GMPR doesn't have ways to go from msas to fips. i hardcoded these because the alternative was to use a private function in gmpr (commented code). not sure which of the two we'd rather have

Reply:
oh shoot, we'd need the weights to invert it wouldn't we.

I think I prefer the private function, but you shouldn't need to generate all 999 PR fips codes, just filter by fips.startswith("72")

Reply:
yea, the weights would be tricky since fips -> msa is an incomplete mapping (not defined for all fips).

got it, good call on `startswith`